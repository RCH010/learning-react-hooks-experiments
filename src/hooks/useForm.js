import { useState } from "react"

export const useForm = (initialState = {}) => {
    const [values, setValues] = useState(initialState);

    const reset= () => {
        setValues(initialState);
    };

    const handleInputChange = ({target}) => {
        setValues({
            ...values,
            [target.name]: target.value
        });
    };
    

    return [values, handleInputChange, reset ];
}

// este custom hook de use form, recibe un objeto, de cada uno de los campos del formulario.