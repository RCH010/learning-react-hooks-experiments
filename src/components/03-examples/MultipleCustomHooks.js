import React from 'react'
import { useFetch } from '../../hooks/useFetch'
import { useCounter } from '../../hooks/useCounter'
import '../02-useEffect/SimpleForm.css'

export const MultipleCustomHooks = () => {
    const {state:id, increment} = useCounter(1);
    const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${id}`);
    // console.log(state);
    const {author, quote} = !!data && data[0];

    return (
        <div>
            <h1 className='h1'>Breaking Bad Quotes</h1>
            <hr />
            {
                loading? (
                    <div className='alert alert-info text-center'>
                        Loading...
                    </div>
                ): (
                <>
                    <div>
                        <blockquote className='blockquote w-50 m-auto'>
                            <p className='mb-0 text-center'>{quote}</p>
                            <footer className='blockquote-footer text-right'>{author}</footer>
                        </blockquote>
                    </div>
                    <div className='d-flex justify-content-end'>
                        <button className='btn btn-success' onClick={increment}>
                            Next Quote
                        </button>
                    </div>
                </>
                )
            }
        </div>
    )
}
