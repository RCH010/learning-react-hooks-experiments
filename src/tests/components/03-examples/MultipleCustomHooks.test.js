import React from 'react';
import { shallow } from "enzyme"
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks"
import { useFetch } from "../../../hooks/useFetch"
import { useCounter } from '../../../hooks/useCounter';
jest.mock("../../../hooks/useFetch")
jest.mock("../../../hooks/useCounter")


describe('Tests in MultipleCustomHooks', () => {
    beforeEach(()=> {
        useCounter.mockReturnValue({
            increment: () => {},
            state: 1
        });
    })

    test('should dsiplay correctly', () => { 
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />)
        expect(wrapper).toMatchSnapshot();
    })

    test('should display info when loaded', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'Raúl Castellanos',
                quote: 'Hola mundo'
            }],
            loading: false,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />)
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hola mundo');
        expect(wrapper.find('footer').text().trim()).toBe('Raúl Castellanos');
    })
    
    
})
