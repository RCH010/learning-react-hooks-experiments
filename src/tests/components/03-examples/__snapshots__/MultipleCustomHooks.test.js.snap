// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in MultipleCustomHooks should dsiplay correctly 1`] = `
<div>
  <h1
    className="h1"
  >
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
</div>
`;
